contracts/test/staking/single-asset-staking.js

```diff
@@ -121,6 +121,46 @@ describe("Single Asset Staking", function () {
     );
   });
 
+
+  it("Stake using WithSender with correct rewards", async () => {
+    const { ogn, anna, ognStaking} = await loadFixture(defaultFixture);
+
+    const annaStartBalance = await ogn.balanceOf(anna.address);
+
+    const numStakeAmount = 1;
+    const stakeAmount = ognUnits(numStakeAmount.toString());
+    // 0.085 is the default reward for three months
+    const expectedReward = ognUnits((numStakeAmount * 0.085).toString());
+
+    await expect(ognStaking.connect(anna).stakeWithSender(anna.address, stakeAmount, threeMonth))
+      .to.be.revertedWith("Only token contract can make this call");
+
+    // from franck
+    // This generate the data needed for calling stakeWithSender
+    const interface = ognStaking.interface
+    const fragment = ognStaking.interface.getFunction("stakeWithSender(address,uint256,uint256)")
+    const fnSig = interface.getSighash(fragment)
+    // calling regular stake here because stakeWithSender inserts the caller's address here
```

Comment:
nit: cleanup those comments below. not relevant anymore.

Reply:
yup thanks removed